src/components/search/Filter/Filter.tsx

```diff
@@ -102,43 +149,61 @@ function Filter({
           expandedIndices={indicesExpanded}
           onChange={onAccordionChange}
         >
-          {facets
-            .filter((facet) => facet.type === 'BOOLEAN')
-            .map(({ label, values, key }, index) => (
-              <AccordionItem
-                key={`${label}-${index}`}
-                testId="filter-accordion"
-                isExpanded={indicesExpanded.has(index)}
-                buttonLabel={label}
-              >
-                <UIList>
-                  {values.map((item) => {
-                    const id = `${label}-${item.label}`
-
-                    return (
-                      <li key={id} className="filter__item">
-                        <Checkbox
-                          id={id}
-                          checked={selectedFilters.some(
+          {filteredFacets.map(({ label, values, key }, index) => (
+            <AccordionItem
+              key={`${label}-${index}`}
+              testId="filter-accordion"
+              isExpanded={indicesExpanded.has(index)}
+              buttonLabel={label}
+              ref={(_) => {
+                // Filter current selected facets from API
+                const filteredValues = values.filter(({ selected }) => selected)
+
+                // Ensures only one array item for each accordion's item
+                if (activeFacets.length < filteredFacets.length) {
+                  activeFacets.push({
+                    accordionIndex: index,
+                    facets:
+                      filteredValues.length > 0
+                        ? filteredValues.map(({ value }) => value)
+                        : [],
```

Comment:
I think this ternary is not necessary. If you map an empty array you will get an empty array.